source/agora/test/Base.d

```diff
@@ -714,11 +714,25 @@ public class TestAPIManager
         static assert (isInputRange!Idxs);
 
         this.setTimeFor(height);
-        foreach (idx; clients_idxs)
-            retryFor(clients[idx].getBlockHeight() == height, timeout,
-                format("Node %s (%s) has block height %s. Expected: %s",
-                    idx, this.nodes[idx].address, clients[idx].getBlockHeight(), height),
-                file, line);
+
+        try
+        {
+            foreach (idx; clients_idxs)
+                retryFor(clients[idx].getBlockHeight() == height, timeout / 2,
+                    format("Node %s (%s) has block height %s. Expected: %s",
+                        idx, this.nodes[idx].address, clients[idx].getBlockHeight(), height),
+                        file, line);
```

Comment:
How about making a nested function for this part as we may have duplicated code lines?

Reply:
Good idea, will do.

Reply:
Updated